add_me_to_your_PATH/sysimage.jl

```diff
@@ -13,6 +13,9 @@ if project.ispackage && !isfile(modulefile)
     end
 end
 
+# Skip generating a system image when there are no dependencies
+isempty(project.dependencies) && exit(0)
```

Comment:
This short-circuit should probably check whether the `packages` passed to `create_sysimage` in `main` is empty. There will almost always be deps, but not always pinned deps.

Reply:
Oh right, ok.

The only difference would be the case where the only dep is `PackageCompiler`, which is not worth changing what you have.